File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.java
class Solution {
    public int solution(String s) {
        int answer = 0;
        
        int count = 0 ;
        char check = s.charAt(0);
        
        
        for(char c : s.toCharArray()){
            if(count == 0){
                check = c;
                answer++;
            }
            if(check == c){
                count++;
            }else{
                count--;
            }
            
        }
        
        return answer;
    }
}

File: C#/프로그래머스/1/140108. 문자열 나누기/문자열 나누기.cs
using System;

public class Solution {
    public int solution(string s) {
int answer = 0;
            int cnt = 0, prev = 0, next = 0;
            char check = ' ';
            while (s.Length > cnt) 
            {
                char c = s[cnt];

                if (check == ' ')
                {
                    check = c;
                    prev++;
                }
                else if (check == c) prev++;
                else if (check != c) next++;
                

                if(prev == next)
                {
                    answer++;
                    prev = 0; next = 0; check = ' ';
                }
                cnt++;
            }
            if (prev > 0) answer++;
            return answer;
    }
}